tests/Authenticator.hs

```diff
@@ -49,17 +55,52 @@ data AuthenticatorSignatureCounter
   = Unsupported
   | Global M.SignatureCounter
   | PerCredential (Map.Map M.CredentialId M.SignatureCounter)
-  deriving (Show)
+
+data SigningBehaviour
+  = GenerateCorrectSignature
+  | GenerateRandomSignature
+
+data RpIdHashBehaviour
+  = GenerateCorrectHash
+  | GenerateRandomHash
+
+-- | Causes bit 6 of the [authenticator data](https://www.w3.org/TR/webauthn-2/#flags) to be set (in)correctly.
+-- We do not have a constructor where both are set wrong, because a test should
+-- already fail if attestation goes wrong.
+data AuthenticatorDataFlagBehaviour
+  = GenerateCorrectFlags
+  | -- | Don't set bit 6 during attestation
+    GenerateWrongFlagAttestation
+  | -- | Set bit 6 during assertion
+    GenerateWrongFlagAssertion
+
+-- Options with a distinct correct value are Maybe values, a Nothing value will
+-- assume the presumed correct behaviour is desired.
+data AuthenticatorConfiguration = AuthenticatorConfiguration
+  { -- | The initial value of which ever SignatureCounter belongs to this credential.
+    -- Only has effect for a PerCredential counter.
+    aCounterInitial :: Maybe M.SignatureCounter,
+    -- | The update function for which ever SignatureCounter belongs to this credential.
+    -- Applied to both the Global and the PerCredential counter
+    -- Default is ((+) 1)
+    aCounterUpdate :: Maybe (M.SignatureCounter -> M.SignatureCounter),
```

Comment:
If we make this an enum, we can have an easy `Arbitrary` instance for `Authenticator`